packages/data/src/world/mq/deku_tree_mq.yml

```diff
@@ -2,126 +2,282 @@
   events:
     MQ_DEKU_WATER_TORCHES: "has_fire"
   locations:
     "MQ Deku Tree After Water Platform Chest": "can_play_time"
-    "MQ Deku Tree Before Water Platform Chest": "true"
     "MQ Deku Tree Grass Spike Room Back 1": "can_cut_grass"
     "MQ Deku Tree Grass Spike Room Back 2": "can_cut_grass"
-    "MQ Deku Tree Grass Larvae Room 1": "can_cut_grass && event(MQ_DEKU_WATER_TORCHES)" #This is placed here because you don't need the souls to get the grass like you do to reach the backrooms.
-    "MQ Deku Tree Grass Larvae Room 2": "can_cut_grass && event(MQ_DEKU_WATER_TORCHES)" #This is placed here because you don't need the souls to get the grass like you do to reach the backrooms.
-    "MQ Deku Tree Grass Spike Room Front 1": "can_cut_grass"
-    "MQ Deku Tree Grass Spike Room Front 2": "can_cut_grass"
-    "MQ Deku Tree Grass Spike Room Front 3": "can_cut_grass"
-"Deku Tree Backrooms":
+"MQ Deku Tree Water Room Door to Room After Water Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Water Room Back": "true"
+    "MQ Deku Tree Room After Water Room Door to Water Room": "event(MQ_DEKU_WATER_TORCHES)"
+"MQ Deku Tree Room After Water Room Door to Water Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Water Room Door to Room After Water Room": "event(DEKU_MQ_ROOM_AFTER_WATER_CLEAR)"
+    "MQ Deku Tree Room After Water Room": "true"
+"MQ Deku Tree Room After Water Room":
   dungeon: DT
   exits:
-    "Deku Tree Water Room Back": "soul_keese && soul_deku_scrub && soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && (has_weapon || has_ranged_weapon_child)" #Judgment call, this fight is way too big for sticks and nuts to be forced.
-    "Deku Tree Basement Ledge": "(is_child && can_use_sticks) || can_use_din"
+    "MQ Deku Tree Room After Water Room Door to Water Room": "true"
+    "MQ Deku Tree Room After Water Room Door to Gravestone Room": "true"
+  events:
+    DEKU_MQ_ROOM_AFTER_WATER_CLEAR: "soul_keese && soul_deku_scrub && soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && (has_weapon || has_ranged_weapon_child)" #Judgment call, this fight is way too big for sticks and nuts to be forced.
+  locations:
+    "MQ Deku Tree Grass Larvae Room 1": "can_cut_grass"
+    "MQ Deku Tree Grass Larvae Room 2": "can_cut_grass"
+"MQ Deku Tree Room After Water Room Door to Gravestone Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room After Water Room": "true"
+    "MQ Deku Tree Gravestone Room Door to Room After Water Room": "event(DEKU_MQ_ROOM_AFTER_WATER_CLEAR)"
+"MQ Deku Tree Gravestone Room Door to Room After Water Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Room After Water Room Door to Gravestone Room": "true"
+    "MQ Deku Tree MQ Grave Room": "true"
+"MQ Deku Tree MQ Grave Room":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Gravestone Room Door to Room After Water Room": "true"
+    "MQ Deku Tree Back Side Room Entry Door": "can_use_sticks || has_fire"
+    "MQ Deku Tree Basement Ledge": "is_child && (can_use_sticks || can_use_din)"
```

Comment:
i'd make an event for this web blocking the crawlspace. Very possible you could use bow to burn it then come back as child.. Likewise, bow alone is fine for the side room (but does not need an event). So both can just be `can_use_sticks || can_use_din || can_use_bow`
Unless child is too short to just move to the side and shoot, then he would need elegy or fire arrows

Reply:
Added webs event; bow alone works for child.